### PR TITLE
Feature/only boss can delete contact

### PR DESCRIPTION
**Pull request by schuetzcarl**
Restrict contact deletion to "Boss" user role.

### CODE REVIEW

File: app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactsCreationTest.kt [hunk 1]
```diff
@@ -156,7 +160,9 @@ class ContactsCreationTest {
         notesEditText.perform(replaceText("This is a test note."))
 
         onView(withId(R.id.confirm_form)).perform(click())
-        Thread.sleep(3000)
```
Comment (TimVW7) [inline app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactsCreationTest.kt:159]: I think I put the delay here, you should be able to take it off, it was for debugging purposes.

File: app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragmentTest.kt [hunk 2]
```diff
@@ -76,11 +75,29 @@ class ContactDetailsFragmentTest {
     }
 
     @Test
-    fun deleteButtonIsDisplayed() {
+    fun deleteButtonIsDisplayedABoss() {
```
Comment (TimVW7) [inline app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragmentTest.kt:78]: You should change to either ...AsBoss or ...ForABoss.

File: app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragment.kt [hunk 3]
```diff
@@ -91,6 +94,9 @@ class ContactDetailsFragment : Fragment() {
             it.findNavController()
                 .navigate(R.id.action_contactDetailsFragment2_to_directoryFragment)
         }
+        if (userViewModel.loggedInUser.value?.role != Role.BOSS) {
+            deleteContactButton.visibility = View.GONE
+        }
```
Comment (TimVW7) [inline app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragment.kt:100]: Might make sense to put this into the initialiseAllButtons function.

**Review by TimVW7 — APPROVED** [review]
Nice code and cleanup, just go over my comments before merging 👍 . Might also be good to improve coverage but it isn't crucial.

File: app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragmentTest.kt (hunk 2 above)
Comment (schuetzcarl) [reply app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragmentTest.kt:78]: Yes my bad forgot the For

**Pull request merged**